In [1]:
#Importing libraries
import pandas as pd
import numpy as np
import os

In [2]:
df=pd.read_csv(r'/Users/prateeksharma/Documents/House prices/02 Data/Original Data/Housing price.csv',index_col=False, encoding='latin-1')

In [3]:
#Displaying first 5 rows of data set
df.head()

City    Price                 Address  Number_Beds  Number_Baths  \
0  Toronto   779900    318 -20 SOUTHPORT ST            3             2   
1  Toronto   799999    818 -60 SOUTHPORT ST            3             1   
2  Toronto   799900  714 -859 THE QUEENSWAY            2             2   
3  Toronto  1200000        275 MORTIMER AVE            4             2   
4  Toronto   668800    420 -388 RICHMOND ST            1             1   

  Province  Population  Latitude  Longitude  Median_Family_Income  
0  Ontario     5647656   43.7417   -79.3733                 97000  
1  Ontario     5647656   43.7417   -79.3733                 97000  
2  Ontario     5647656   43.7417   -79.3733                 97000  
3  Ontario     5647656   43.7417   -79.3733                 97000  
4  Ontario     5647656   43.7417   -79.3733                 97000

In [4]:
#Displaying last 5 rows of data set
df.tail()

City    Price                     Address  Number_Beds  \
35763  Halifax   439900  301 60 Walter Havill Drive            2   
35764  Halifax  6990000      1160 Rockcliffe Street            5   
35765  Halifax   419900  212 60 Walter Havill Drive            2   
35766  Halifax   949900           10 Idlewylde Road            3   
35767  Halifax   592900     208 2842-2856 Gottingen            2   

       Number_Baths     Province  Population  Latitude  Longitude  \
35763             2  Nova Scotia      431479   44.8857    63.1005   
35764             5  Nova Scotia      431479   44.8857    63.1005   
35765             2  Nova Scotia      431479   44.8857    63.1005   
35766             1  Nova Scotia      431479   44.8857    63.1005   
35767             1  Nova Scotia      431479   44.8857    63.1005   

       Median_Family_Income  
35763                 86753  
35764                 86753  
35765                 86753  
35766                 86753  
35767                 86753

In [5]:
#Checking dimensions
df.shape

(35768, 10)

In [6]:
df.describe()

Price   Number_Beds  Number_Baths    Population      Latitude  \
count  3.576800e+04  35768.000000  35768.000000  3.576800e+04  35768.000000   
mean   9.432963e+05      3.283661      2.532403  6.360151e+05     47.446556   
std    1.020110e+06      1.730654      1.371910  1.120016e+06      3.333855   
min    2.150000e+04      0.000000      0.000000  6.338200e+04     42.283300   
25%    4.599000e+05      2.000000      2.000000  1.091670e+05     43.866700   
50%    6.990000e+05      3.000000      2.000000  2.424600e+05     49.025000   
75%    1.095000e+06      4.000000      3.000000  5.228880e+05     49.888100   
max    3.700000e+07    109.000000     59.000000  5.647656e+06     53.916900   

          Longitude  Median_Family_Income  
count  35768.000000          35768.000000  
mean     -98.421636          89643.103416  
std       22.280935          12132.353510  
min     -123.936400          62400.000000  
25%     -122.316700          82000.000000  
50%     -104.606700          89000.000000  
75%      -79.866700          97000.000000  
max       63.100500         133000.000000

109 Beds and 59 Baths in a house is a bit unsual and i'll be dropping them as its a very values like that in the dataset.

In [7]:
# Drop rows where the number of beds is more than 30
df = df[df['Number_Beds'] <= 30]

# Drop rows where the number of baths is more than 20
df = df[df['Number_Baths'] <= 20]

In [8]:
df.describe()

Price   Number_Beds  Number_Baths    Population      Latitude  \
count  3.575900e+04  35759.000000  35759.000000  3.575900e+04  35759.000000   
mean   9.421679e+05      3.273665      2.527112  6.360882e+05     47.447599   
std    1.016660e+06      1.563360      1.314913  1.120143e+06      3.333626   
min    2.150000e+04      0.000000      0.000000  6.338200e+04     42.283300   
25%    4.599000e+05      2.000000      2.000000  1.091670e+05     43.866700   
50%    6.990000e+05      3.000000      2.000000  2.424600e+05     49.025000   
75%    1.095000e+06      4.000000      3.000000  5.228880e+05     49.888100   
max    3.700000e+07     30.000000     20.000000  5.647656e+06     53.916900   

          Longitude  Median_Family_Income  
count  35759.000000          35759.000000  
mean     -98.426185          89643.530943  
std       22.281892          12133.439775  
min     -123.936400          62400.000000  
25%     -122.316700          82000.000000  
50%     -104.606700          89000.000000  
75%      -79.866700          97000.000000  
max       63.100500         133000.000000

In [9]:
#Displaying column names
df.columns

Index(['City', 'Price', 'Address', 'Number_Beds', 'Number_Baths',
       'Province', 'Population', 'Latitude', 'Longitude',
       'Median_Family_Income'],
      dtype='object')

In [10]:
#Renaming 'City' to City
df.rename(columns={'City': 'City'}, inplace=True)

In [11]:
#Displaying column names
df.columns

Index(['City', 'Price', 'Address', 'Number_Beds', 'Number_Baths', 'Province',
       'Population', 'Latitude', 'Longitude', 'Median_Family_Income'],
      dtype='object')

In [12]:
#Dropping address column as its not needed for the analysis
df1=df.drop(columns = ['Address'])

In [13]:
df1

City    Price  Number_Beds  Number_Baths     Province  Population  \
0      Toronto   779900            3             2      Ontario     5647656   
1      Toronto   799999            3             1      Ontario     5647656   
2      Toronto   799900            2             2      Ontario     5647656   
3      Toronto  1200000            4             2      Ontario     5647656   
4      Toronto   668800            1             1      Ontario     5647656   
...        ...      ...          ...           ...          ...         ...   
35763  Halifax   439900            2             2  Nova Scotia      431479   
35764  Halifax  6990000            5             5  Nova Scotia      431479   
35765  Halifax   419900            2             2  Nova Scotia      431479   
35766  Halifax   949900            3             1  Nova Scotia      431479   
35767  Halifax   592900            2             1  Nova Scotia      431479   

       Latitude  Longitude  Median_Family_Income  
0       43.7417   -79.3733                 97000  
1       43.7417   -79.3733                 97000  
2       43.7417   -79.3733                 97000  
3       43.7417   -79.3733                 97000  
4       43.7417   -79.3733                 97000  
...         ...        ...                   ...  
35763   44.8857    63.1005                 86753  
35764   44.8857    63.1005                 86753  
35765   44.8857    63.1005                 86753  
35766   44.8857    63.1005                 86753  
35767   44.8857    63.1005                 86753  

[35759 rows x 9 columns]

In [14]:
#Checking for missing values
df1.isnull().sum()

City                    0
Price                   0
Number_Beds             0
Number_Baths            0
Province                0
Population              0
Latitude                0
Longitude               0
Median_Family_Income    0
dtype: int64

No missing values 

In [15]:
#Finding Duplicates
df_dups=df1[df.duplicated()]

In [16]:
df_dups

City    Price  Number_Beds  Number_Baths          Province  \
637           Toronto   599900            2             1           Ontario   
1037          Toronto   929000            3             2           Ontario   
1314        Vancouver   799900            2             2  British Columbia   
1369        Vancouver  1898000            3             4  British Columbia   
1389        Vancouver   790000            2             2  British Columbia   
...               ...      ...          ...           ...               ...   
35672  Trois-Rivieres   925000            3             2            Quebec   
35673  Trois-Rivieres   347000            5             2            Quebec   
35674  Trois-Rivieres   279900            5             2            Quebec   
35675  Trois-Rivieres   272000            2             1            Quebec   
35676  Trois-Rivieres   519485            4             2            Quebec   

       Population  Latitude  Longitude  Median_Family_Income  
637       5647656   43.7417   -79.3733                 97000  
1037      5647656   43.7417   -79.3733                 97000  
1314      2426160   49.2500  -123.1000                 90000  
1369      2426160   49.2500  -123.1000                 90000  
1389      2426160   49.2500  -123.1000                 90000  
...           ...       ...        ...                   ...  
35672      139163   46.3500   -72.5500                 62400  
35673      139163   46.3500   -72.5500                 62400  
35674      139163   46.3500   -72.5500                 62400  
35675      139163   46.3500   -72.5500                 62400  
35676      139163   46.3500   -72.5500                 62400  

[2516 rows x 9 columns]

2516 rows are duplicated which is about 7% of the data set and since they are mostly coming up like this because of same features like number_beds and number_baths which is why i'll be keeping these values.

In [17]:
# Exporting data set in pickle format
df1.to_pickle(r'/Users/prateeksharma/Documents/House prices/02 Data/Prepared Data/df_clean.pkl')